src/lib/core/observe-content/observe-content.ts

```diff
@@ -20,12 +35,26 @@ export class ObserveContent implements AfterContentInit, OnDestroy {
   private _observer: MutationObserver;
 
   /** Event emitted for each change in the element's content. */
-  @Output('cdkObserveContent') event = new EventEmitter<void>();
+  @Output('cdkObserveContent') event = new EventEmitter<MutationRecord[]>();
 
-  constructor(private _elementRef: ElementRef) {}
+  /** Used for debouncing the emitted values to the observeContent event. */
+  private _debouncer = new Subject<MutationRecord[]>();
+
+  /** Debounce interval for emitting the changes. */
+  @Input() debounce: number;
+
+  constructor(
+    private _mutationObserverFactory: MdMutationObserverFactory,
+    private _elementRef: ElementRef) { }
 
   ngAfterContentInit() {
-    this._observer = new MutationObserver(mutations => mutations.forEach(() => this.event.emit()));
+    this._debouncer
+      .debounceTime(this.debounce)
```

Comment:
Can we conditionally add `debounceTime` to the chain if `debounce !== 0`? Andrew noticed when working on the scroll dispatcher that `auditTime` incurred a cost even when it was set to zero.